sockeye/arguments.py

```diff
@@ -1085,6 +1085,22 @@ def add_training_args(params):
                               action="store_true",
                               help='In addition to keeping the last n params files, also keep params from checkpoint 0.')
 
+    
+    train_params.add_argument('--cache-last-best-params',
+                              type=int,
+                              default=4,
```

Comment:
What would this default need to be set to, to not change the default behavior of the training CLI as it is right now?

Reply:
I am probably missing something, but why do we need this argument if we already have `--keep-last-params'? And why use 'cache' instead of 'keep'?

Reply:
The default value would be anything <= the value of `--keep-last-params`. 

The reason for adding the arg `--cache-last-best-params` in addition to `--keep-last-params' is because the semantics are different. `--keep-last-params'  does just what it says, makes sure that we keep only the last N params files we saved. However, these last N params files in time are not necessarily the last N **best** params files, and very often are not. If we don't make the effort to keep the N **best** params files, then if we want to build an averaged params file, we will not be able to use the true N best. 

Now, we could just set `--keep-last-params' to zero, and keep every params file we save off. But this can eat an awful lot of disk space. This change makes sure we keep the information we need to build an accurate averaged parameter set, and to save disk space while so doing.

There is no great reason for using 'cache' instead of 'keep'. It seemed more evocative of what this is trying to do, and to distinguish of `--keep-last-params`.

Reply:
Got you, thanks for the explanation! Let's set the default value to match previous behavior. How about adding a small note in the help string of this new argument that in the 'worst case' this would keep up to x additional param files on top of the y files from `--keep-last-params`.

Reply:
Just saw your recent commit that you already added a modified help string, thanks!